### ISSUE TITLE
Linking libosrm v5.3.3 using pkg-config

### ISSUE THREAD

**Issue by jcoupey** [Project-OSRM/osrm-backend#2901]
I'm linking my client code to `libosrm` by providing additional flags [based on `pkg-config` output](https://github.com/VROOM-Project/vroom/blob/release-v1.0.0/src/makefile#L12-L15).

Everything is fine with v5.1.0, as it adds the expected flags:
```
$ pkg-config --cflags libosrm
-I/usr/local/include -I/usr/local/include/osrm
```

Now the same with v5.3.3 provides:
```
$ pkg-config --cflags libosrm
-DBOOST_TEST_DYN_LINK -DBOOST_SPIRIT_USE_PHOENIX_V3 -DBOOST_RESULT_OF_USE_DECLTYPE -DBOOST_FILESYSTEM_NO_DEPRECATED -flto=4 -Wall -Wextra -pedantic -Wuninitialized -Wunreachable-code -Wstrict-overflow=1 -D_FORTIFY_SOURCE=2 -fdiagnostics-color=auto -fPIC -ffunction-sections -fdata-sections -std=c++11 -fopenmp -I/usr/local/include -I/usr/local/include/osrm -I/[my local osrm folder]/third_party/libosmium/include -I/usr/include/luabind -I/usr/include/lua5.2
```

So I start getting `error: use of ‘auto’ in lambda parameter declaration only available with -std=c++14 or -std=gnu++14`. Probably because my own `-std=c++14` is overridden by the `-std=c++11` provided by `libosrm` cflags.

Then if I keep the same build command using all above flags except `-std=c++11`, I get errors regarding the calls to functions in the `osrm` namespace for code that use to compile when linking to `libosrm` v5.1.0.
```
$ g++ -std=c++14 -Wextra -Wpedantic -Wall -O3 -DBOOST_LOG_DYN_LINK -DBOOST_TEST_DYN_LINK -DBOOST_SPIRIT_USE_PHOENIX_V3 -DBOOST_RESULT_OF_USE_DECLTYPE -DBOOST_FILESYSTEM_NO_DEPRECATED -flto=4 -Wall -Wextra -pedantic -Wuninitialized -Wunreachable-code -Wstrict-overflow=1 -D_FORTIFY_SOURCE=2 -fdiagnostics-color=auto -fPIC -ffunction-sections -fdata-sections -fopenmp -I/usr/local/include -I/usr/local/include/osrm -I[my local osrm folder]/third_party/libosmium/include -I/usr/include/luabind -I/usr/include/lua5.2 -D LIBOSRM=true -c main.cpp -o main.o
In file included from main.cpp:24:0:
./loaders/libosrm_loader.h: In member function ‘virtual matrix<unsigned int> libosrm_loader::get_matrix() const’:
./loaders/libosrm_loader.h:51:76: error: no matching function for call to ‘osrm::util::FloatLongitude::FloatLongitude(const double&)’
       params.coordinates.push_back({osrm::util::FloatLongitude(location.lon),
                                                                            ^
In file included from /usr/local/include/osrm/util/typedefs.hpp:31:0,
                 from /usr/local/include/osrm/engine/phantom_node.hpp:32,
                 from /usr/local/include/osrm/engine/hint.hpp:31,
                 from /usr/local/include/osrm/engine/api/base_parameters.hpp:32,
                 from /usr/local/include/osrm/engine/api/route_parameters.hpp:31,
                 from /usr/local/include/osrm/route_parameters.hpp:31,
                 from ./loaders/libosrm_loader.h:15,
                 from main.cpp:24:
/usr/local/include/osrm/util/coordinate.hpp:51:1: note: candidate: osrm::util::FloatLongitude::FloatLongitude()
 OSRM_STRONG_TYPEDEF(double, FloatLongitude)
 ^
/usr/local/include/osrm/util/coordinate.hpp:51:1: note:   candidate expects 0 arguments, 1 provided
/usr/local/include/osrm/util/coordinate.hpp:51:1: note: candidate: constexpr osrm::util::FloatLongitude::FloatLongitude(const osrm::util::FloatLongitude&)
/usr/local/include/osrm/util/coordinate.hpp:51:1: note:   no known conversion for argument 1 from ‘const double’ to ‘const osrm::util::FloatLongitude&’
/usr/local/include/osrm/util/coordinate.hpp:51:1: note: candidate: constexpr osrm::util::FloatLongitude::FloatLongitude(osrm::util::FloatLongitude&&)
/usr/local/include/osrm/util/coordinate.hpp:51:1: note:   no known conversion for argument 1 from ‘const double’ to ‘osrm::util::FloatLongitude&&’
[...]
```

Any idea on how to overcome this?

**Comment by daniel-j-h** [comment]
@springmeyer you changed the pkg-config scripts a couple of month ago, could you look into this?

**Comment by jcoupey** [comment]
Just reporting that I get the same behaviour with `libosrm` v5.2.8. 

Note for future self: check my code against any new rc...

**Comment by daniel-j-h** [comment]
Yeah @jcoupey looks like https://github.com/Project-OSRM/osrm-backend/commit/6ecc123d1535182eaa1f97f40dbfac8d89f96e75 to me.

**Comment by jcoupey** [comment]
With `libosrm` v5.4.0-rc.1, the `-std=c++11` has been replaced with `-std=c++1y` so I get the second error reported in the initial post.

**Comment by jcoupey** [comment]
The second reported error had nothing to do with compilation flags and is now solved (https://github.com/VROOM-Project/vroom/issues/43#issuecomment-250719221).

This leaves only the problem of using c++14 standards with `libosrm` v5.3.3, which *will* be ok in v5.4 (I checked with a rc version).

Up to you to close this!

**Comment by daniel-j-h** [comment]
Sorry for all the inconveniences here :( Closing this as resolved in v5.4.
